Add owners to SHOW commands

ajstorm opened the pull request:
Main drop of code to add owner information to SHOW DATABASES, TABLES, ENUMS, and SCHEMAS.  Includes all testcase changes as well.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54708)
<!-- Reviewable:end -->

solongordon reviewed (changes requested):
> Reviewed 2 of 4 files at r1, 5 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54708)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm)

___
*[pkg/sql/pg_catalog.go, line 1230 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwcK278VcAV6BB01AQ:-MHwcK282vsnGogwYMoB:bxnmre0) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/pg_catalog.go#L1230)):*
> ```Go
> 	datname Name,
> 	datdba OID,
> 	owner Name,
> ```

I think we should avoid adding this column if possible since it's not present in Postgres. Can you join `datdba` to `pg_roles` instead?
___
*[pkg/sql/delegate/show_databases.go, line 22 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwbhq0BJtiGx4trofR:-MHwbhq0BJtiGx4trofS:b6xp1d1) ([raw file](https://github.com/cockroachdb/cockroach/blob/915cc0067c05b54155c10d78485fee8163e6d1b6/pkg/sql/delegate/show_databases.go#L22)):*
> ```Go
> func (d *delegator) delegateShowDatabases(stmt *tree.ShowDatabases) (tree.Statement, error) {
> 	query := `SELECT
> 	datname AS database_name, databaseowner as database_owner
> ```

Why change the `name` column to `datname`?
___
*[pkg/sql/delegate/show_databases.go, line 21 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdFpu9t5kByeIyPDy:-MHwdFpu9t5kByeIyPDz:b-cb57z2) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L21)):*
> ```Go
> 
> func (d *delegator) delegateShowDatabases(stmt *tree.ShowDatabases) (tree.Statement, error) {
> 	// The 'node' role is internal-only (not exposed in the pg_roles table), so we have to hard-code its value here
> ```

I'm curious if there's a good reason we don't put the node user in `pg_roles`, but that's probably not worth messing with here.
___
*[pkg/sql/delegate/show_databases.go, line 24 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwcWgK9iuIdRBvrAzu:-MHwcWgK9iuIdRBvrAzv:b56fhd3) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L24)):*
> ```Go
> 	query := `SELECT
> 	datname AS database_name,
>   CASE WHEN d.datdba = 3233629770
> ```

Could you make the OID a query parameter which is passed in rather than just a magic number? Then you can set it to `UserOid(security.NodeUser)` and ensure that it's correct even if our OID hashing logic changes.
___
*[pkg/sql/delegate/show_databases.go, line 25 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdC9E8XqGi0UFP-W2:-MHwdC9F5Wl26fcUeTns:b-b4z8se) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L25)):*
> ```Go
> 	datname AS database_name,
>   CASE WHEN d.datdba = 3233629770
>   THEN 'node'
> ```

Could also pass in `security.NodeUser` here.
___
*[pkg/sql/delegate/show_databases.go, line 28 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwcUKB9PuXrnbQ0O1t:-MHwcUKB9PuXrnbQ0O1u:b-maypku) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L28)):*
> ```Go
>   ELSE rl.rolname
>   END
>   AS database_owner
> ```

 I'd prefer just `owner` rather than `database_owner`.
___
*[pkg/sql/delegate/show_enums.go, line 18 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwbv2UA7iwWx4vMC5p:-MHwbv2UA7iwWx4vMC5q:bqyosqg) ([raw file](https://github.com/cockroachdb/cockroach/blob/915cc0067c05b54155c10d78485fee8163e6d1b6/pkg/sql/delegate/show_enums.go#L18)):*
> ```Go
> 	query := `
> SELECT
> 	schema, name, owner, string_agg(label, '|') AS value
> ```

Nit: Might be nicer if the `owner` column came last.
___
*[pkg/sql/delegate/show_enums.go, line 29 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdTB8DUK8uFsxhEjk:-MHwdTB8DUK8uFsxhEjl:b-soz0r7) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_enums.go#L29)):*
> ```Go
> 			pg_catalog.pg_enum AS enum
> 			JOIN pg_catalog.pg_type AS type ON (type.oid = enum.enumtypid)
>       JOIN pg_catalog.pg_roles AS rl on (type.typowner = rl.oid)
> ```

I'd say make this a LEFT JOIN just in case there isn't a matching entry in `pg_roles` for some reason.
___
*[pkg/sql/delegate/show_schemas.go, line 30 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwddIH-KY-P0dGH_-K:-MHwddIH-KY-P0dGH_-L:b-3sakhq) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_schemas.go#L30)):*
> ```Go
> 	}
> 	getSchemasQuery := fmt.Sprintf(`
>       SELECT nspname AS schema_name, rolname AS schema_owner
> ```

Again I'd just call the owner column `owner`.
___
*[pkg/sql/delegate/show_tables.go, line 55 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdi5G9sfNx6nWLWl5:-MHwdi5G9sfNx6nWLWl6:b7difgo) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_tables.go#L55)):*
> ```Go
>        ELSE 'table'
>        END AS type,
>        rl.rolname AS table_owner,
> ```

`owner`
___
*[pkg/sql/delegate/show_tables.go, line 59 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdlX93U2zq63GDgtc:-MHwdlX93U2zq63GDgtd:b-14k65l) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_tables.go#L59)):*
> ```Go
>        %[3]s
>   FROM %[1]s.pg_catalog.pg_class AS pc
>   JOIN %[1]s.pg_catalog.pg_roles AS rl on (pc.relowner = rl.oid)
> ```

Prefer a LEFT JOIN here.


<!-- Sent from Reviewable.io -->

ajstorm reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54708)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @solongordon)

___
*[pkg/sql/pg_catalog.go, line 1230 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwcK278VcAV6BB01AQ:-MHwhWOh3H1ZO6ooS0CW:b-e6hige) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/pg_catalog.go#L1230)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

I think we should avoid adding this column if possible since it's not present in Postgres. Can you join `datdba` to `pg_roles` instead?
</blockquote></details>

Agreed.  This was from the original attempt and I missed removing it in the second attempt.  Will be removed in next PR.
___
*[pkg/sql/delegate/show_databases.go, line 22 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwbhq0BJtiGx4trofR:-MHwi0_hBYcul4LU4idY:b-clu5lk) ([raw file](https://github.com/cockroachdb/cockroach/blob/915cc0067c05b54155c10d78485fee8163e6d1b6/pkg/sql/delegate/show_databases.go#L22)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

Why change the `name` column to `datname`?
</blockquote></details>

I'm querying the pg_catalog.pg_database table now instead of the crdb_internal.databases table because pg_database has the datdba column.  If you think we should still be querying the crdb_internal.databases table I can do that, but I'll need to join with the pg_database table to get the datdba column.  Seemed more straightforward the new way.
___
*[pkg/sql/delegate/show_databases.go, line 24 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwcWgK9iuIdRBvrAzu:-MHwo3e08g3g7CqQbAAn:b-vujr0g) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L24)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

Could you make the OID a query parameter which is passed in rather than just a magic number? Then you can set it to `UserOid(security.NodeUser)` and ensure that it's correct even if our OID hashing logic changes.
</blockquote></details>

As discussed over Slack, this should be more to your liking now (and mine too).  Thanks for all of the help steering this in the right direction.
___
*[pkg/sql/delegate/show_databases.go, line 25 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdC9E8XqGi0UFP-W2:-MHwoVhe4_6uEXJk_cwd:b-ucfkn4) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L25)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

Could also pass in `security.NodeUser` here.
</blockquote></details>

Thanks.
___
*[pkg/sql/delegate/show_databases.go, line 28 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwcUKB9PuXrnbQ0O1t:-MHwoYjJ61NrXQn0LGEz:b-c3msi1) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L28)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

 I'd prefer just `owner` rather than `database_owner`.
</blockquote></details>

ok, done.
___
*[pkg/sql/delegate/show_enums.go, line 18 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwbv2UA7iwWx4vMC5p:-MHxg92NDYSFb8aMQSeM:b-ou6x6) ([raw file](https://github.com/cockroachdb/cockroach/blob/915cc0067c05b54155c10d78485fee8163e6d1b6/pkg/sql/delegate/show_enums.go#L18)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

Nit: Might be nicer if the `owner` column came last.
</blockquote></details>

sure.  Done
___
*[pkg/sql/delegate/show_enums.go, line 29 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdTB8DUK8uFsxhEjk:-MHxgLg-AtdKYptvbnte:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_enums.go#L29)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

I'd say make this a LEFT JOIN just in case there isn't a matching entry in `pg_roles` for some reason.
</blockquote></details>

Done.
___
*[pkg/sql/delegate/show_schemas.go, line 30 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwddIH-KY-P0dGH_-K:-MHxgTLx53-kswhmN-qv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_schemas.go#L30)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

Again I'd just call the owner column `owner`.
</blockquote></details>

Done.
___
*[pkg/sql/delegate/show_tables.go, line 55 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdi5G9sfNx6nWLWl5:-MHxgZnl67JTId75ZxkE:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_tables.go#L55)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

`owner`
</blockquote></details>

Done
___
*[pkg/sql/delegate/show_tables.go, line 59 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwdlX93U2zq63GDgtc:-MHxgdU0DJAxKXfVgDFk:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_tables.go#L59)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

Prefer a LEFT JOIN here.
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

solongordon reviewed (changes requested):
Looking good, on to the tests!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-:-MI-heCpABD3h9HXqM41:bury7mu)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm and @solongordon)

___
*[pkg/sql/pg_catalog.go, line 18 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MI-gwCwE7y__V0o5FED:-MI-gwCwE7y__V0o5FEE:b-uzix71) ([raw file](https://github.com/cockroachdb/cockroach/blob/585c313d6bbb9104fce97ba4829157046e26582a/pkg/sql/pg_catalog.go#L18)):*
> ```Go
> 	"encoding/binary"
> 	"fmt"
> 	"github.com/cockroachdb/cockroach/pkg/sql/delegate"
> ```

I recommend configuring your editor to run `crlfmt` (our enhanced version of `go fmt`) on save. More details here: https://cockroachlabs.slack.com/archives/C4VM697PC/p1540908498001700
___
*[pkg/sql/pg_catalog.go, line 55 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MI-h8CFBLuggt6QhK1A:-MI-h8CFBLuggt6QhK1B:b-uintpx) ([raw file](https://github.com/cockroachdb/cockroach/blob/585c313d6bbb9104fce97ba4829157046e26582a/pkg/sql/pg_catalog.go#L55)):*
> ```Go
> 
> 	passwdStarString = tree.NewDString("********")
> ```

Nit: Leftover blank line
___
*[pkg/sql/delegate/show_databases.go, line 22 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwbhq0BJtiGx4trofR:-MI-eroPDxgKR0Beb5c9:b-w5jbvf) ([raw file](https://github.com/cockroachdb/cockroach/blob/915cc0067c05b54155c10d78485fee8163e6d1b6/pkg/sql/delegate/show_databases.go#L22)):*
<details><summary><i>Previously, ajstorm wrote…</i></summary><blockquote>

I'm querying the pg_catalog.pg_database table now instead of the crdb_internal.databases table because pg_database has the datdba column.  If you think we should still be querying the crdb_internal.databases table I can do that, but I'll need to join with the pg_database table to get the datdba column.  Seemed more straightforward the new way.
</blockquote></details>

Sorry I totally misread this. I thought you had changed the display name.
___
*[pkg/sql/delegate/show_databases.go, line 24 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MHwcWgK9iuIdRBvrAzu:-MI-fq2Y016aeaVfcsfB:bwowdkq) ([raw file](https://github.com/cockroachdb/cockroach/blob/75d552012e12b92d2378f804f2feb29c5c632d0b/pkg/sql/delegate/show_databases.go#L24)):*
<details><summary><i>Previously, ajstorm wrote…</i></summary><blockquote>

As discussed over Slack, this should be more to your liking now (and mine too).  Thanks for all of the help steering this in the right direction.
</blockquote></details>

Yes, great! Nit: I'd prefer `%d` over `%v` here for specificity. It enforces that we expect an int param. You'll generally see `%v` used more in test code, log and error output, etc.


<!-- Sent from Reviewable.io -->

knz reviewed (commented):
got pinged via the cli change, but not reviewing this. Just a "build system" comment below.

> Reviewed 2 of 14 files at r10, 1 of 2 files at r12, 1 of 2 files at r15.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-:-MIOLWUa81hrxfkV3wOM:beh9s1l)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm, @dt, and @solongordon)

___
*[go.sum, line 73 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MIOLDtx1eXZnXBPLMLW:-MIOLDty5dipvc9ytWRR:b-t4lf4z) ([raw file](https://github.com/cockroachdb/cockroach/blob/274ee234ea962b6edd92bbfc33fc3202d583b1f5/go.sum#L73)):*
> ```
> github.com/PuerkitoBio/goquery v1.5.0 h1:uGvmFXOA73IKluu/F84Xd1tt/z07GYm8X49XKHP7EJk=
> github.com/PuerkitoBio/goquery v1.5.0/go.mod h1:qD2PgZ9lccMbQlc7eEOjaeRlFQON7xY8kdmcsrnKqMg=
> github.com/PuerkitoBio/goquery v1.5.1 h1:PSPBGne8NIUWw+/7vFBV+kG2J/5MOjbzc7154OaKCSE=
> ```

1) Is it possible you forgot to run `go mod tidy`

2) I don't fully understand why go.mod / go.sum needed to change in this PR. Did they? (If not you can revert these two -- otherwise, you'll need to update `vendor`)


<!-- Sent from Reviewable.io -->

ajstorm commented:
@knz Thanks for catching that.  Those files were inadvertently changed after I ran rebase.  At this point I don't see why I'd need to change them, so I'll try reverting and see if that works.

solongordon reviewed (changes requested):
Whew, looks good to me! Would be nice to have @raduberinde eyeball the `execbuilder` test change since I don't know what's expected there.

Would you squash your commits into one and then I can give this a thumbs up?

> Reviewed 3 of 5 files at r4, 1 of 7 files at r5, 1 of 12 files at r6, 8 of 17 files at r7, 9 of 14 files at r10, 2 of 2 files at r11, 2 of 2 files at r14, 4 of 6 files at r18, 2 of 3 files at r19, 1 of 1 files at r20, 1 of 2 files at r21, 1 of 1 files at r23.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-:-MIUI4uo2xVdESN0NmGs:be34x4f)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm, @RaduBerinde, and @solongordon)

___
*[pkg/sql/opt/optbuilder/testdata/delegate, line 1 at r23](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MIUF37K2GIBrSm1LAil:-MIUF37K2GIBrSm1LAim:b-5ts3mr) ([raw file](https://github.com/cockroachdb/cockroach/blob/91d33265c3e802cf0cb66e78217ba3ef394413e6/pkg/sql/opt/optbuilder/testdata/delegate#L1)):*
> ```
> # Note: t is the default database for the test catalog.
> ```

Was this file deleted on purpose?


<!-- Sent from Reviewable.io -->

ajstorm reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54708)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz, @RaduBerinde, and @solongordon)

___
*[go.sum, line 73 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MIOLDtx1eXZnXBPLMLW:-MIUUsIu0fNGjlHNf9zu:b-go0j53) ([raw file](https://github.com/cockroachdb/cockroach/blob/274ee234ea962b6edd92bbfc33fc3202d583b1f5/go.sum#L73)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

1) Is it possible you forgot to run `go mod tidy`

2) I don't fully understand why go.mod / go.sum needed to change in this PR. Did they? (If not you can revert these two -- otherwise, you'll need to update `vendor`)
</blockquote></details>

Removed these changes.  They were the result of a bad merge.
___
*[pkg/sql/pg_catalog.go, line 18 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MI-gwCwE7y__V0o5FED:-MIUV07S7MvZ-sYgKkP4:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/585c313d6bbb9104fce97ba4829157046e26582a/pkg/sql/pg_catalog.go#L18)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

I recommend configuring your editor to run `crlfmt` (our enhanced version of `go fmt`) on save. More details here: https://cockroachlabs.slack.com/archives/C4VM697PC/p1540908498001700
</blockquote></details>

Done.
___
*[pkg/sql/opt/optbuilder/testdata/delegate, line 1 at r23](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-MIUF37K2GIBrSm1LAil:-MIUVnD7-zLxFdhRZVkQ:bqqr51z) ([raw file](https://github.com/cockroachdb/cockroach/blob/91d33265c3e802cf0cb66e78217ba3ef394413e6/pkg/sql/opt/optbuilder/testdata/delegate#L1)):*
<details><summary><i>Previously, solongordon (Solon) wrote…</i></summary><blockquote>

Was this file deleted on purpose?
</blockquote></details>

Yes, the testcase was added to execbuilder instead.


<!-- Sent from Reviewable.io -->

ajstorm committed:
sql: Change SHOW DATABASES, ENUMS, SCHEMAS, TABLES to include owner

Previously, the SHOW SQL statements listed above did not include owner
information. As a result, a user interested in the owner of each asset
would have to join with some external table (typically a pg_catalog)
table. This change makes ownership more obvious by including it
directly in the SHOW statements.

Release note (sql change): Added owner column to SHOW DATABASES, ENUMS, TABLES,
SCHEMAS

RaduBerinde commented:
opt test changes LGTM!

solongordon reviewed (approved):
One last thing: the convention on Cockroach commits is to prefix the commit message with the primary package which is affected. So in this case the commit title would be `sql: add owner to SHOW commands`. Other than that looks good to go!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54708#-:-MIVjCaH-lRUc30vNYZr:b-1vdo1d)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz, @RaduBerinde, and @solongordon)



<!-- Sent from Reviewable.io -->

ajstorm commented:
bors +r

ajstorm commented:
bors r+

otan commented on pkg/sql/delegate/show_enums.go:
```diff
@@ -15,24 +15,26 @@ import "github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	query := `
 SELECT
-	schema, name, string_agg(label, '|') AS value
+	schema, name, string_agg(label, '|'), owner AS value
```
i think this should be:
`	schema, name, string_agg(label,) as value, owner`

otan replied:
Fix in #55139 !

ajstorm replied:
Yeah, good catch.  Not sure how I missed that :-(

otan replied:
easy to miss if logic test aren't printing out the column names :P
